synapse/handlers/room.py

```diff
@@ -662,13 +662,20 @@ async def create_room(
 
         directory_handler = self.hs.get_handlers().directory_handler
         if room_alias:
-            await directory_handler.create_association(
-                requester=requester,
-                room_id=room_id,
-                room_alias=room_alias,
-                servers=[self.hs.hostname],
-                check_membership=False,
+            # Check if publishing is blocked by a third party module
+            allowed_by_third_party_rules = await (
+                self.third_party_event_rules.check_room_can_be_added_to_public_rooms_directory(
+                    room_id
+                )
             )
+            if allowed_by_third_party_rules:
+                await directory_handler.create_association(
```

Comment:
I don't understand why this association is done before checking `is_publishing_room_allowed` ans why this is not done only if `is_public` is true
@anoadragon453 a sync about this may be useful for me

Reply:
Ah, you and rich came to the [same conclusion](https://github.com/matrix-org/synapse/pull/8292#discussion_r487066679). Yes, that's a mistake.